### DISCUSSION TITLE
Feature request - Ability to create routes that take in email via SMTP, and webhook post it to an endpoint and reply via HTTP request

### DISCUSSION THREAD

**Discussion by samip5** [postalserver/postal#2116] [Feature suggestions]
I would love to see a way in Postal to create an route that is sent though via webhook eg to a Matrix channel and that you could reply to via an HTTP request.

**Comment by willpower232** [comment]
I'm not sure what you mean by "Matrix channel" but Postal already provides the ability to route incoming email via HTTP and accepts messages being sent via HTTP so you could probably make an application using these features to satisfy your requirement.

**Reply by samip5** [reply]
> I'm not sure what you mean by "Matrix channel" but Postal already provides the ability to route incoming email via HTTP and accepts messages being sent via HTTP so you could probably make an application using these features to satisfy your requirement.

Currently Postal doesn't allow me to customize the payload to specific spec, which Matrix spec requires (https://spec.matrix.org/v1.3/).


Currently the payload is:

POST request to `https://homeserver-address/_matrix/client/r0/rooms/<room-id>/send/m.room.message?access_token=<token>`

```
msg = {
  "body": <message-body>,
  "html": <html-formatted-body>,
  "msgtype": "m.text"
}
```

**Reply by willpower232** [reply]
Ah I understand, if you're aiming to customise which room it ends up in then I imagine the fastest thing to do would be to make your own "middleware" that can receive Postals [incoming message](https://docs.postalserver.io/developer/http-payloads) and pass it on to the right Matrix room with the correct format.

Customising Postal to suit would be a lot of very specific work so probably not desired.
